06.wc/wc.rb

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def exec
+  options = parse_options
+  options = { c: true, l: true, w: true } if options.empty?
+  paths = ARGV
+  paths.empty? ? display_word_count_from_stdin(options) : display_word_count_from_files(options, paths)
+end
+
+def parse_options
+  opt = OptionParser.new
+  options = {}
+  opt.on('-c')
+  opt.on('-l')
+  opt.on('-w')
+  opt.parse!(ARGV, into: options)
+  options
+end
+
+def display_word_count_from_stdin(options)
+  lines = $stdin.readlines
+  word_count_map = build_word_count_map(lines)
+  display_word_count(word_count_map, options)
+end
+
+def build_word_count_map(lines)
+  {
+    number_of_lines: count_line(lines),
+    number_of_words: count_word(lines),
+    bytesize: count_bytesize(lines)
+  }
+end
+
+def count_line(lines)
+  lines.size
+end
+
+def count_word(lines)
+  lines.sum { |line| line.split(/[ \t\n]+/).size }
+end
+
+def count_bytesize(lines)
+  lines.sum(&:bytesize)
+end
+
+def display_word_count(word_count_map, options)
+  word_counts = []
+  word_counts << format_word_count(word_count_map[:number_of_lines]) if options[:l]
+  word_counts << format_word_count(word_count_map[:number_of_words]) if options[:w]
+  word_counts << format_word_count(word_count_map[:bytesize]) if options[:c]
+  word_counts << " #{word_count_map[:path]}" if word_count_map.key?(:path)
+  puts word_counts.join
+end
+
+def format_word_count(word_count)
+  word_count.to_s.rjust(8)
+end
+
+def display_word_count_from_files(options, paths)
```

Comment:
```suggestion
def display_word_count_from_files(paths, options)
```

optionsはあくまでオプションなので、引数の順番としては最後に持ってくることが多いですね